docs/vcpkg-schema-definitions.schema.json

```diff
@@ -0,0 +1,715 @@
+        }
+      },
+      "patternProperties": {
+        "^\\$": {}
+      },
+      "required": [
+        "description"
+      ],
+      "additionalProperties": false
+    },
+    "filesystem-registry": {
+      "description": "A port registry backed by a git repo.",
+      "type": "object",
+      "properties": {
+        "kind": {
+          "type": "string",
+          "pattern": "filesystem"
+        },
+        "path": {
+          "type": "string",
+          "description": "The location on the filesystem where the registry is rooted."
+        }
+      },
+      "required": [
+        "kind",
+        "path"
+      ]
+    },
+    "info-block": {
+      "type": "object",
+      "properties": {
+        "version": {
+          "$ref": "#/definitions/semantic-version"
+        },
+        "id": {
+          "$ref": "#/definitions/slash-identifier"
+        },
+        "summary": {
+          "description": "A shorter description of the artifact.",
+          "type": "string"
+        },
+        "priority": {
+          "type": "integer",
+          "minimum": 0,
+          "default": 0
+        },
+        "description": {
+          "type": "string"
+        },
+        "options": {
+          "$ref": "#/definitions/string-array",
+          "default": []
+        }
+      },
+      "additionalProperties": false,
+      "required": [
+        "version",
+        "id"
+      ]
+    },
+    "install": {
+      "if": {
+        "type": "object"
+      },
+      "then": {
+        "$ref": "#/definitions/install-instance"
+      },
+      "else": {
+        "type": "array",
+        "items": {
+          "$ref": "#/definitions/install-instance"
+        }
+      }
+    },
+    "install-instance": {
+      "type": "object",
+      "oneOf": [
+        {
+          "properties": {
+            "unzip": {
+              "$ref": "#/definitions/uri-or-uris"
+            },
+            "sha256": {
+              "$ref": "#/definitions/sha256"
+            },
+            "sha512": {
+              "$ref": "#/definitions/sha512"
+            },
+            "strip": {
+              "type": "integer",
+              "minimum": 0,
+              "default": 0
+            },
+            "transform": {
+              "type": "string",
+              "description": "An SED expression to run on the file names before extracting them."
+            },
+            "lang": {
+              "type": "string",
+              "description": "The locale under which this install should be applied."
+            },
+            "nametag": {
+              "type": "string",
+              "description": "A string to add to the name of the artifact when extracted on disk."
+            }
+          },
+          "required": [
+            "unzip"
+          ],
+          "additionalProperties": false
+        },
+        {
+          "properties": {
+            "untar": {
+              "$ref": "#/definitions/uri-or-uris"
+            },
+            "sha256": {
```

Comment:
It might be clearer to refactor these common properties out since they're shared between `unzip`/`untar`/`nupkg`

Reply:
I tried to do that but because they are not allowed for git I didn't see an easy way to do it.